### PR TITLE
Cleanup world

### PR DESCRIPTION
**Pull request by Alberth289346**
Move some data and code out of world.

### CODE REVIEW

**Comment by Alberth289346** [comment]
editroom still needs an afterload for the new wall data

File: CorsixTH/Lua/world.lua [hunk 1]
```diff
@@ -905,13 +836,13 @@ function World:previousSpeed()
   end
 end
 
--- Return if the selected speed the same as the current speed.
+--! Return true if the given speed the same as the current speed.
```
Comment (lewri) [inline CorsixTH/Lua/world.lua:839]: It's `!return` for luadoc isn't it?

**Review by lewri — COMMENTED** [review]
Did you need to do something in edit room still btw, or is it not important to this pr?

**Comment by Alberth289346** [comment]
Done and confirmed it works. All set now

File: CorsixTH/Lua/dialogs/edit_room.lua [hunk 2]
```diff
@@ -90,6 +93,58 @@ function UIEditRoom:UIEditRoom(ui, room_type)
   self:registerKeyHandlers()
 end
 
+function UIEditRoom:initWallTypes()
```
Comment (TheCycoONE) [inline CorsixTH/Lua/dialogs/edit_room.lua:96]: One bad place to another... but it's only used here, so I accept.

File: CorsixTH/Lua/world.lua (hunk 1 above)
Comment (TheCycoONE) [reply CorsixTH/Lua/world.lua:839]: Yes, but convention is to prefer the method summary to the return parameter description when doing both would be redundant. The change is good.

**Review by TheCycoONE — APPROVED** [review]

File: CorsixTH/Lua/world.lua (hunk 1 above)
Comment (Alberth289346) [reply CorsixTH/Lua/world.lua:839]: @lewri: I just aimed at getting the line recognized in luadoc, as it seemed mostly complete although I agree not entirely according to luadoc.

**Review by lewri — APPROVED** [review]
Give or take my comment, looks fine.

**Pull request merged**